reporting_dsl.go

```diff
@@ -27,6 +27,10 @@ CurrentSpecReport returns information about the current running spec.
 The returned object is a types.SpecReport which includes helper methods
 to make extracting information about the spec easier.
 
+During construction of the test tree, the returned report provides
+information about the current container node. Fields related to
+a leaf node are not set in that case.
```

Comment:
This is up for debate. It's convenient because it reuses much of the existing API (function and type) and infrastructure (global entry in suite), but it's also a bit confusing because it's not perfectly clear what is being set in the result.